DQM/CSCMonitorModule/plugins/CSCDQM_StripClusterFinder.cc

```diff
@@ -189,84 +191,75 @@ namespace cscdqm {
-        IC2MAX = MEStripClusters[ic2].IRTBNDStrip;
-        JC2MIN = MEStripClusters[ic2].LFTBNDTime;
-        JC2MAX = MEStripClusters[ic2].IRTBNDTime;
-        if ((IC2MIN >= IC1MIN && IC2MIN <= IC1MAX && JC2MIN >= JC1MIN && JC2MIN <= JC1MAX) ||
-            (IC2MIN >= IC1MIN && IC2MIN <= IC1MAX && JC2MAX >= JC1MIN && JC2MAX <= JC1MAX) ||
-            (IC2MAX >= IC1MIN && IC2MAX <= IC1MAX && JC2MIN >= JC1MIN && JC2MIN <= JC1MAX) ||
-            (IC2MAX >= IC1MIN && IC2MAX <= IC1MAX && JC2MAX >= JC1MIN && JC2MAX <= JC1MAX)) {
-          KillCluster();
+    uint32_t icstart = 0;  //!!!???
+    for (uint32_t ic1 = icstart; ic1 < MEStripClusters.size(); ic1++) {
+      C1 c1;
+      c1.IC1MIN = MEStripClusters[ic1].LFTBNDStrip;
+      c1.IC1MAX = MEStripClusters[ic1].IRTBNDStrip;
+      c1.JC1MIN = MEStripClusters[ic1].LFTBNDTime;
+      c1.JC1MAX = MEStripClusters[ic1].IRTBNDTime;
+      for (uint32_t ic2 = ic1 + 1; ic2 < MEStripClusters.size(); ic2++) {
+        C2 c2;
+        c2.IC2MIN = MEStripClusters[ic2].LFTBNDStrip;
+        c2.IC2MAX = MEStripClusters[ic2].IRTBNDStrip;
+        c2.JC2MIN = MEStripClusters[ic2].LFTBNDTime;
+        c2.JC2MAX = MEStripClusters[ic2].IRTBNDTime;
+        if ((c2.IC2MIN >= c1.IC1MIN && c2.IC2MIN <= c1.IC1MAX && c2.JC2MIN >= c1.JC1MIN && c2.JC2MIN <= c1.JC1MAX) ||
+            (c2.IC2MIN >= c1.IC1MIN && c2.IC2MIN <= c1.IC1MAX && c2.JC2MAX >= c1.JC1MIN && c2.JC2MAX <= c1.JC1MAX) ||
+            (c2.IC2MAX >= c1.IC1MIN && c2.IC2MAX <= c1.IC1MAX && c2.JC2MIN >= c1.JC1MIN && c2.JC2MIN <= c1.JC1MAX) ||
+            (c2.IC2MAX >= c1.IC1MIN && c2.IC2MAX <= c1.IC1MAX && c2.JC2MAX >= c1.JC1MIN && c2.JC2MAX <= c1.JC1MAX)) {
+          icstart = KillCluster(ic1, ic2, c1, c2);
           return true;
         } else {
-          if ((IC1MIN >= IC2MIN && IC1MIN <= IC2MAX && JC1MIN >= JC2MIN && JC1MIN <= JC2MAX) ||
-              (IC1MIN >= IC2MIN && IC1MIN <= IC2MAX && JC1MAX >= JC2MIN && JC1MAX <= JC2MAX) ||
-              (IC1MAX >= IC2MIN && IC1MAX <= IC2MAX && JC1MIN >= JC2MIN && JC1MIN <= JC2MAX) ||
-              (IC1MAX >= IC2MIN && IC1MAX <= IC2MAX && JC1MAX >= JC2MIN && JC1MAX <= JC2MAX)) {
-            KillCluster();
+          if ((c1.IC1MIN >= c2.IC2MIN && c1.IC1MIN <= c2.IC2MAX && c1.JC1MIN >= c2.JC2MIN && c1.JC1MIN <= c2.JC2MAX) ||
+              (c1.IC1MIN >= c2.IC2MIN && c1.IC1MIN <= c2.IC2MAX && c1.JC1MAX >= c2.JC2MIN && c1.JC1MAX <= c2.JC2MAX) ||
+              (c1.IC1MAX >= c2.IC2MIN && c1.IC1MAX <= c2.IC2MAX && c1.JC1MIN >= c2.JC2MIN && c1.JC1MIN <= c2.JC2MAX) ||
+              (c1.IC1MAX >= c2.IC2MIN && c1.IC1MAX <= c2.IC2MAX && c1.JC1MAX >= c2.JC2MIN && c1.JC1MAX <= c2.JC2MAX)) {
+            icstart = KillCluster(ic1, ic2, c1, c2);
             return true;
           }
         }
       }
     }
     return false;
   }
-  void StripClusterFinder::KillCluster(void) {
+  uint32_t StripClusterFinder::KillCluster(uint32_t ic1, uint32_t ic2, C1 const& c1, C2 const& c2) {
```

Comment:
This respect the structure of the original code and updates `icstart`.
However, there is no need to have this method `uint`, not to update `icstart` inside the loop, when it was only used to set the initial value. This is also notified by the SA, which signals a dead assignment here.
I don't want to be too picky: let merge this PR as such now (i.e. same flow as in the original code). However, I think that in a follow up PR once this is merged one could revert this method being `void` and avoid updating `icstart` after its usage as starting point for the loop.